research/gaia/gremlin/gremlin_core/src/structure/graph.rs

```diff
@@ -33,52 +40,78 @@ impl<E: Element + Send + Sync> Default for QueryParams<E> {
     }
 }
 
+impl<E: Element + Send + Sync> FromPb<Option<pb::QueryParams>> for QueryParams<E> {
+    fn from_pb(query_params_pb: Option<pb::QueryParams>) -> Result<Self, ParseError>
+    where
+        Self: Sized,
+    {
+        let mut query_params = QueryParams::default();
+        if let Some(query_params_pb) = query_params_pb {
```

Comment:
I am suggesting the following codes:
```
Implement the follow functions<where xxx is labels, limit, required_properties, predicates, extra_params> in QueryParams: 
fn with_xxx(mut self, Option<xxx>) -> Self
```
Then in from_pb, you can do:
```
let mut query_params = QueryParams::default();
if let Some(query_params_pb) = query_params_pb {
      query_params = query_params.with_labels_from_pb(labels)
        .with_limit(limit)
        .with_required_properties(xx)
        .with_predicates(xx)
        .with_extra_params(xx);
}
```

Reply:
Done